example/lib/cubit/sort_cubit/sort_cubit.dart

```diff
@@ -0,0 +1,390 @@
+import 'package:equatable/equatable.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:native_filters/native_filters.dart';
+import 'package:rxdart/rxdart.dart';
+
+part 'sort_state.dart';
+
+class SortCubit extends Cubit<SortState> {
+  SortCubit() : super(SortInitial());
+
+  @override
+  Stream<SortState> get stream => super.stream.doOnListen(() {
+        if (state is SortInitial) {
+          fetchSortData(const FilterFactory());
+        }
+      });
+
+  Future<void> fetchSortData(FilterFactory filtersFactory) async {
```

Comment:
Init FiltersFactory into constructor

Reply:
Init FiltersFactory into main and pass List into cubit